### PR TITLE
Integration test

### PR DESCRIPTION
**Pull request by t-bello7**
- Write integration tests with Capybara gem
- Secure app from n+1 problems with bullet gem

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -2,12 +2,12 @@ class PostsController < ApplicationController
   def index
     @posts = Post.where(author_id: params[:user_id])
     @user = User.find(params[:user_id])
```
Comment (FrederickMih) [inline app/controllers/posts_controller.rb]: - [Optional] In order to keep our application scalable, let's kindly eliminate the N+1 query problem from our app. Here we can't fetch the comments of a user from the db without fetching together all the posts which can lead to performance issues.


I think `@posts = @user.posts.includes(:comments)` can resolve the issue

File: spec/rails_helper.rb [hunk 2]
```diff
@@ -32,7 +34,6 @@
 RSpec.configure do |config|
   # Remove this line if you're not using ActiveRecord or ActiveRecord fixtures
   config.fixture_path = "#{:Rails.root}/spec/fixtures"
```
Comment (FrederickMih) [inline spec/rails_helper.rb]: - [ ] Hi @t-bello7, for some reason your tests were not passing probably cause by this line of code:
Screenshot from 2022-12-15 06-05-46

- After I comment the line of code, the tests ran successfully:


Screenshot from 2022-12-15 06-09-27

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: @t-bello7 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


Screenshot from 2022-10-20 09-38-51


## Highlights :green_circle: 

- All the different tests scenarios were covered using Capybara :+1:
- PR is descriptive :+1: 
- All Linter checks passed :heavy_check_mark: 


## Required Changes ♻️

N/A

_Check the comments under the review._

## Optional suggestions
N/A
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (t-bello7) [reply app/controllers/posts_controller.rb]: I was able to eliminate n+1 queries by adding include to the views but I would make changes as this is better

File: spec/rails_helper.rb (hunk 2 above)
Comment (t-bello7) [reply spec/rails_helper.rb]: I get this error when I push to GitHub. Do you have a suggestion on how I can fix it?
Screenshot from 2022-12-15 06-35-06

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (t-bello7) [reply app/controllers/posts_controller.rb]: I get eager loading detected when I add this line of code 
`@posts = @user.posts.includes(:comments)`
Screenshot from 2022-12-15 06-40-40

**Review by Davidosky007 — APPROVED** [review]
Hi @t-bello7 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* No linter errors 👍 
* Well written PR description 👍 
* Good looking readme file 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**